### DISCUSSION TITLE
Handling stderr from a client

### DISCUSSION THREAD

**Discussion by spoutn1k** [Eugeny/russh#258] [Q&A] [answered]
Hello,

I am following the examples and am confused as to how to get the contents of standard error from a remotely executed command.

I suspected `ChannelMsg::ExtendedData` to return extra file descriptors contents but it failed to return anything when running a command that outputs to stderr. `ChannelMsg::Data` was also empty.

Thanks !

**Comment by Eugeny** [comment, marked as answer]
You're on the right track - stderr data will arrive as `ExtendedData` with `ext=1`.

As an example, you can add this to the event loop in the `client_exec_simple` example:

```
                ChannelMsg::ExtendedData { ref data , ..} => {
                    print!("<stderr>:");
                    stdout.write_all(data).await?;
                    stdout.flush().await?;
                }
```

and it will work:

```
cargo run --all-features --example client_exec_simple -- --private-key ... 192.168... bash -c "echo stderr > /dev/stderr; echo stdout > /dev/stdout"
   Compiling russh v0.43.0-beta.1 (/Users/eugene/Work/russh/russh)
    Finished dev [unoptimized + debuginfo] target(s) in 0.99s
[2024-03-04T13:24:19Z INFO  client_exec_simple] Connecting to ...:22
[2024-03-04T13:24:19Z INFO  client_exec_simple] Key path: ...
[2024-03-04T13:24:20Z INFO  client_exec_simple] Connected
<stderr>:stderr
stdout
Exitcode: 0
```

**Reply by spoutn1k** [reply]
Aaargh feels bad I should have thought of checking id 1 ! I was running with `ext=2` by pure force of bash habit. Thank you for the quick answer !
